### PR TITLE
Refactor all HTTP requests into client classes

### PR DESCRIPTION
**Pull request by JayH5**
Supersedes #35, comparing against `develop` this time.

### CODE REVIEW

**Comment by JayH5** [comment]
ping :bell:

File: consular/clients.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        """
+        d = self.request('GET', path, **kwargs)
+        return d.addCallback(self._response_json_if_ok)
+
+    def _response_json_if_ok(self, response):
+        """
+        Get the response JSON content if the respones code is OK (200), else
+        raise an `UnexpectedResponseError`.
+        """
+        if response.code == OK:
```
Comment (smn) [inline consular/clients.py:92]: The full range of `2xx` is essentially OK, does `OK` cover these?

Comment (smn) [inline consular/clients.py:92]: It doesn't look like it http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.4.0/twisted/web/_responses.py

**Comment by smn** [comment]
:+1:

**Pull request merged**